woocommerce/woocommerce-gateway-stripe
# Fix ECE button displayed on synced subscription variations

Fixes #3607 

## Changes proposed in this Pull Request:
ECE buttons should be hidden if the subscription is synchronized. In variable subscriptions with synced and non-synced variations, ECE button still appears when switched to a synced variation. This PR fixes the issue by checking if `getSelectedProductData` response has an error. 


## Testing instructions

- Create a variable subscription product with 2 variations (Blue and Red) and a sign up free.
- Make the Blue variation synchronized (i.e on 10th day).
- Enable the ECE feature flag.
- As a shopper go to the product page.
- Notice that, in `develop` branch the ECE button is displayed for both variations.
- Confirm that in this branch the ECE button is displayed only for the unsynced variation (Red).
